main.py

```diff
@@ -0,0 +1,50 @@
+import logging
+
+import uvicorn
+from fastapi import FastAPI
+from mujtaba_charm.utils.sample import hello
+
+app = FastAPI()
+
+logging.basicConfig(
+    level=logging.INFO, format="[%(asctime)s] %(levelname)s in %(module)s: %(message)s"
+)
+logger = logging.getLogger(__name__)
+
+
+@app.get("/hello")
+@app.get("/hello/{name}")
+async def get_hello(name: str = None):
+    """
+    Prints a greeting message.
+
+    If a name parameter is provided, it greets that name; otherwise, it greets the world.
+
+    Args:
+        name (str, optional): The name to greet. Defaults to None.
+
+    Returns:
+        dict: A dictionary containing a greeting message.
+    """
+    if name:
+        logger.info(f"Handling /hello/{name} request")
+        message = hello(name)
+    else:
+        logger.info("Handling /hello request")
+        message = hello()
```

Comment:
remove code duplication

explore python's `or` operator to avoid the if/else

```python
hello(name or "my-dear-user")
```

Reply:
Done